### ISSUE TITLE
Consider adopting BCP 47 language codes (e.g. en-US) in lieu of POSIX ones (e.g. en_US)

### ISSUE THREAD

**Issue by AndreaBarbasso** [DSpace/DSpace#11448]
## Is your feature request related to a problem? Please describe.
Looking at https://github.com/DSpace/dspace-angular/issues/435 's PR https://github.com/DSpace/dspace-angular/pull/4776, a11y tests fails because we can't use POSIX language codes in HTML5 `lang` attributes, since those require BCP 47 codes.  
We need to understand how POSIX codes are used on the REST project, and then decide if moving to BCP 47 is beneficial for a11y purposes.

## Describe alternatives or workarounds you've considered
I am implementing a small workaround on the front end right now - a very simple string replace (from `_` to `-`) in order to avoid e2e test errors. We will have to remove it if we decide to move to BCP 47 codes.

**Comment by tdonohue** [comment]
This ticket came out of a discussion in yesterday's Developers Meeting: https://wiki.lyrasis.org/display/DSPACE/2025-10-16+DSpace+Developers+Meeting

It may require a volunteer to more closely analyze the impact of changing the format of these language codes.  However, at a glance, it seemed reasonable to Committers (at least those in attendance at the meeting).  We just need to verify that such a change doesn't result in undesired side effects elsewhere (especially in the backend code, or in other machine interfaces like OAI-PMH, etc).

**Comment by mwoodiupui** [comment]
I was concerned with how this might impact our use of [`java.util.Locale`](https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html), but `Locale` uses the BCP 47 format so my concern is satisfied.
